client/my-sites/stats/summary/index.jsx

```diff
@@ -222,22 +234,44 @@ class StatsSummary extends Component {
 				break;
 			case 'annualstats':
 				title = translate( 'Annual site stats' );
+				backLabel = localizedTabNames.insights;
+				backLink = `/stats/insights/`;
 				summaryView = <AnnualSiteStats key="annualstats" />;
 				break;
 		}
 
+		// Append the site domain and period as needed.
+		const isFixedNavHeadersEnabled = config.isEnabled( 'stats/fixed-nav-headers' );
+		const domain = this.props?.path.split( '/' ).pop();
+		if ( domain.length > 0 ) {
+			backLink += domain;
+			const lastChartDate = localStorage.getItem( 'jp-stats-last-chart-date' );
+			if ( lastChartDate.length > 0 ) {
```

Comment:
This is causing a render-breaking bug; here's my repro:

- Open the [live branch](https://calypso.live/?image=registry.a8c.com/calypso/app:build-47524).
- Navigate to /stats/day/example.com
- With the browser console open, click on "Post and Pages" section header.
- Note the console error: ` Cannot read properties of null` on this line.

Here's a potential fix:
```suggestion
			if ( lastChartDate?.length > 0 ) {
```

Reply:
Good catch! I thought I saw this happen but then couldn't reproduce it. I'll get this fixed up.